test/transform/CollapseTable.test.js

```diff
@@ -838,6 +838,44 @@ describe('CollapseTable', () => {
       assert.deepEqual(window.document.getElementById('c').style.display, 'none')
       assert.deepEqual(window.document.getElementById('d').style.display, 'none')
     })
+
+    it('moves a table up to the content block when it is nested', () => {
+      const html = `
+        <div><div class=content_block><div class=not_content_block><div>
+        <table class=infobox>info</table>
+        </div></div></div></div>
+      `
+      const window = domino.createWindow(html)
+      collapseTables(window, window.document)
+      const contentBlock = window.document.querySelector('.content_block')
+      assert.ok(contentBlock)
+      assert.deepEqual(contentBlock.parentNode.parentNode.tagName, 'BODY')
+      const collapsibleContainer = contentBlock.childNodes[0]
+      assert.ok(collapsibleContainer)
+      assert.ok(collapsibleContainer.classList.contains('pagelib_collapse_table_container'))
+      const table = collapsibleContainer.querySelector('.infobox')
+      assert.ok(table)
+      assert.deepEqual(table.parentNode, collapsibleContainer)
+    })
+
+    it('moves a table to the tag with the data-mw-section-id attribute when it is nested', () => {
+      const html = `
+        <div><section data-mw-section-id=0><div class=not_content_block><div>
```

Comment:
To make this test more realistic: Parsoid doesn't have an outer div around the `<section>` tag.

Reply:
Fixed